src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpTransportFactory.php

```diff
@@ -32,7 +32,10 @@ public function __construct(SerializerInterface $serializer = null)
 
     public function createTransport(string $dsn, array $options): TransportInterface
     {
-        return new AmqpTransport(Connection::fromDsn($dsn, $options), $this->serializer);
+        $serializer = $options['serializer'] ?? $this->serializer;
+        unset($options['serializer']);
```

Comment:
This was the easiest (no interface) way to make this change happen. But, it IS a bit ugly. We could change the interface to `createTransport(string $dsn, array $options, SerializerInterface $serializer = null)` - the serializer *does* seem a bit unique.

Reply:
Glad I found this comment 😄 

I think that the serializer option is quite unique... but breaking the BC on this class for such an argument means that we can't really afford to have a second "unique" case like this. I'd vote for a `createTransport(string $dsn, array $options, SerializerInterface $serializer)` (not sure why it needs to be nullable as anyway it breaks the BC) but we need more input from the @symfony/deciders there 🤔